src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -0,0 +1,207 @@
+    private static class Noop extends ColumnConstraints
+    {
+        private Noop()
+        {
+            super(Collections.emptyList());
+        }
+
+        @Override
+        public void validate(ColumnMetadata columnMetadata)
+        {
+            // Do nothing. It is always valid
+        }
+    }
+
+    public final static class Raw
+    {
+        private final List<ColumnConstraint<?>> constraints;
+
+        public Raw(List<ColumnConstraint<?>> constraints)
+        {
+            this.constraints = constraints;
+        }
+
+        public Raw()
+        {
+            this.constraints = Collections.emptyList();
+        }
+
+        public ColumnConstraints prepare()
+        {
+            if (constraints.isEmpty())
+                return NO_OP;
+            return new ColumnConstraints(constraints);
+        }
+    }
+
+    public static class Serializer implements MetadataSerializer<ColumnConstraints>
+    {
+
+        @Override
+        public void serialize(ColumnConstraints columnConstraint, DataOutputPlus out, Version version) throws IOException
+        {
+            out.writeInt(columnConstraint.getSize());
+            for (ColumnConstraint constraint : columnConstraint.getConstraints())
+            {
+                // We serialize the serializer ordinal in the enum to save space
+                out.writeShort(constraint.getConstraintType().ordinal());
+                constraint.serializer().serialize(constraint, out, version);
+            }
+        }
+
+        @Override
+        public ColumnConstraints deserialize(DataInputPlus in, Version version) throws IOException
+        {
+            List<ColumnConstraint<?>> columnConstraints = new ArrayList<>();
+            int numberOfConstraints = in.readInt();
+            for (int i = 0; i < numberOfConstraints; i++)
+            {
+                int serializerPosition = in.readShort();
+                ColumnConstraint<?> constraint = (ColumnConstraint<?>) ConstraintType
+                                                                       .getSerializer(serializerPosition)
+                                                                       .deserialize(in, version);
+                columnConstraints.add(constraint);
+            }
+            return new ColumnConstraints(columnConstraints);
+        }
+
+        @Override
+        public long serializedSize(ColumnConstraints columnConstraint, Version version)
+        {
+            long constraintsSize = TypeSizes.INT_SIZE;
+            for (ColumnConstraint constraint : columnConstraint.getConstraints())
+            {
+                constraintsSize += TypeSizes.SHORT_SIZE;
+                constraintsSize += constraint.serializer().serializedSize(constraint, version);
+            }
+            return constraintsSize;
+        }
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+
+        if (!(o instanceof ColumnMetadata))
```

Comment:
This is checking the wrong class, so it always returns false when comparing 2 ColumnConstraints instances.

Reply:
Good catch!